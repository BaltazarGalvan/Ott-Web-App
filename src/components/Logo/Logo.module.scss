@use '../../styles/variables';
@use '../../styles/theme';
@use '../../styles/mixins/responsive';

//
// Logo
// --------------------------------

.brand {
  align-self: center;
  text-align: center;
}

.logo {
  max-width: 100%;
  max-height: variables.$header-height - 10px;
  max-height: 36px;
  vertical-align: middle;
  cursor: pointer;
}

//
// mediaQueries
// --------------------------------

@include responsive.mobile-and-tablet() {
  .brand {
    flex: 1;

    .logo {
      max-height: variables.$header-height-mobile - 20px;
    }
  }
}

@include responsive.mobile-only() {
  .brand {
    transition: opacity 0.2s ease;
  }
}

@include responsive.desktop-only() {
  .brand {
    width: 180px;
    text-align: left;
  }
}
