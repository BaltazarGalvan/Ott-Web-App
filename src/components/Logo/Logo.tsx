import React from 'react';
import { useHistory } from 'react-router-dom';

import styles from './Logo.module.scss';

type Props = {
  src: string;
};

const Logo: React.FC<Props> = ({ src }: Props) => {
  const history = useHistory();

  const handleClick = () => {
    history.push('/');
  };

  return (
    <div className={styles.brand}>
      <img className={styles.logo} alt="logo" src={src} onClick={handleClick} />
    </div>
  );
};

export default Logo;
