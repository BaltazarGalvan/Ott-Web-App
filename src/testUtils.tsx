import { BrowserRouter as Router } from 'react-router-dom';
import React, { ReactNode, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

interface WrapperProps {
  children?: ReactNode;
}

export const wrapper = ({ children }: WrapperProps) => <Router>{children}</Router>;

const customRender = (ui: ReactElement, options?: RenderOptions) => render(ui, { wrapper, ...options });

export const mockWindowLocation = (path: string) => {
  Object.defineProperty(window, 'location', {
    value: {
      pathname: path,
      assign: jest.fn(),
    },
  });
};

export const mockMatchMedia = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    })),
  });
};

export { customRender as render };
