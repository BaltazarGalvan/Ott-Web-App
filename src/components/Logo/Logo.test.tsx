import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render } from '../../testUtils';

import Logo from './Logo';

describe('<Logo/>', () => {
  it('renders and matches snapshot', () => {
    const { container } = render(<Logo src="123" onLoad={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });

  it('calls the onLoad function after the image has been loaded', () => {
    const onLoad = jest.fn();
    const { getByAltText } = render(<Logo src="testimage" onLoad={onLoad} />);

    fireEvent.load(getByAltText('logo'));

    expect(onLoad).toBeCalledTimes(1);
  });

  it('calls the onLoad function when the image fails to load', () => {
    const onLoad = jest.fn();
    const { getByAltText } = render(<Logo src="testimage" onLoad={onLoad} />);

    fireEvent.error(getByAltText('logo'));

    expect(onLoad).toBeCalledTimes(1);
  });
});
