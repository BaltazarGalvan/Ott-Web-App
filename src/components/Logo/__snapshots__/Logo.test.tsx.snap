// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> renders and matches snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="logo"
      class="logo"
      src="123"
    />
  </a>
</div>
`;
